Sources/SemanticAnalyzer/SemanticError.swift

```diff
@@ -550,4 +550,30 @@ extension Diagnostic {
     return Diagnostic(severity: .warning, sourceLocation: variableDeclaration.sourceLocation,
                       message: "Variables are already implicitly mutating")
   }
+
+  static func defaultArgumentsNotAtEnd(_ functionDeclaration: FunctionDeclaration) -> Diagnostic {
```

Comment:
nit: since you're only using the sourceLocation, could you not remove the duplication for function declarations and event declarations by making it generic and specifying a constraint on the generic type? Maybe it's not possible